cohttp-eio/src/server.ml

```diff
@@ -95,17 +95,13 @@ let rec handle_request reader writer flow handler =
       if not (is_custom body) then Writer.wakeup writer;
       if Http.Request.is_keep_alive request then
         handle_request reader writer flow handler
-      else Eio.Flow.close flow
-  | (exception End_of_file) | (exception Eio.Net.Connection_reset _) ->
-      Eio.Flow.close flow
```

Comment:
Is this always the case? Is there a risk that this changes in the future or that having an explict close would make it close earlier?

A priori, I would feel safer to always call the close, eventually with a catch to drop the relevant double close errors

Reply:
Yes, it will always close. Eio ensures that FDs cannot be leaked. And it would be an error to close the connection and then continue trying to handle it. In the git version of Eio trying to close it manually here is a compile-time error (and in Eio 0.3, due to a bug, closing it manually results in an error being reported).